Studies/source/GenStudy.cxx

```diff
@@ -0,0 +1,173 @@
+     TH1D_ENTRY(bm_2, 10, -0.5, 9.5)
+     TH1D_ENTRY(bm, 10, -0.5, 9.5)
+     TH1D_ENTRY(bm_VM, 10, -0.5, 9.5)
+     TH1D_ENTRY(reco_0jet_matched_gen_particles, 10, -0.5, 9.5)
+     TH1D_ENTRY(reco_1jet_matched_gen_particles, 10, -0.5, 9.5)
+ };
+
+class GenStudy {
+public:
+    GenStudy(const Arguments& _args) :
+        args(_args), output(root_ext::CreateRootFile(args.outputFile())), anaData(output)
+        {
+            GenEvent::intializeNames(args.particleNameTypeFile());
+        }
+
+    size_t getIndexMaxPt(std::vector<const GenParticle*> visible_daughters){
+        size_t max_pt_index = 0;
+        for (size_t i = 1; i < visible_daughters.size(); i++)
+            max_pt_index = visible_daughters.at(0)->momentum.pt() > visible_daughters.at(i)->momentum.pt() ? 0 : (i);
+        return max_pt_index;
+    }
+
+    void Run()
+    {
+        auto file = root_ext::OpenRootFile(args.inputPath());
+        auto tuple = ntuple::CreateEventTuple(args.treeName(), file.get(), true, ntuple::TreeState::Full);
+        const double deltaR_value = 0.2;
+
+        for(const auto& event : *tuple) {
+           if(event.eventEnergyScale != 0) continue;
+
+            GenEvent genEvent(event);
+//            genEvent.Print();
+            const GenParticleSet higgsPair = genEvent.GetParticles(particles::ParticleCode::higgs);
+            if(higgsPair.size() != 2)
+                throw analysis::exception("Higgs pair size must be 2, insteed has a size of '%1%'") %higgsPair.size(); //Stampare eventId e albero
+
+            const GenParticle* h_tautau = nullptr;
+            const GenParticle* h_bb = nullptr;
+
+            for(const auto& higgs : higgsPair){
+                if(higgs->daughters.size() != 2)
+                throw analysis::exception("Higgs decay products must be 2, insteed has a size of '%1%'") %higgs->daughters.size();
+
+                if(std::abs(higgs->daughters.at(0)->pdg) == particles::ParticleCode::tau &&
+                    std::abs(higgs->daughters.at(1)->pdg) == particles::ParticleCode::tau)
+                    h_tautau = higgs;
+                else if(std::abs(higgs->daughters.at(0)->pdg) == particles::ParticleCode::b &&
+                    std::abs(higgs->daughters.at(1)->pdg) == particles::ParticleCode::b)
+                     h_bb = higgs;
+                 else
+                     throw analysis::exception("Higgs pair doesn't decay in a pair of taus or b jets h1 -> '%1%', h2 -> '%2%'.") %higgs->daughters.at(0)->pdg %higgs->daughters.at(1)->pdg;
+            }
+            if(h_tautau == nullptr || h_bb == nullptr)
+                throw analysis::exception("The Higgs pairs are not filled");
+            if(h_tautau->daughters.size() != 2 || h_bb->daughters.size() != 2)
+                throw analysis::exception("Each of the Higgs pairs must decay in two particles.");
+
+            int n_tau_matches = 0;
+            for (size_t tau_index = 0; tau_index < h_tautau->daughters.size(); tau_index++) {
```

Comment:
create a separate method to check acceptance and decay channel.
- if at least one tau (or one b when it will be implemented) is outside is acceptance -> continue to the next event
- same if di-tau decay mode doesn't match the channel.